Changes for getting a successful build on Solaris.  Not suitable for upstream

--- a/js/src/gc/Memory.cpp	2015-09-17 13:12:18.000000000 -0700
+++ b/js/src/gc/Memory.cpp	2017-07-06 15:07:30.630983096 -0700
@@ -83,7 +83,7 @@
     return uintptr_t(p) % alignment;
 }
 
-void*
+static void*
 TestMapAlignedPagesLastDitch(size_t size, size_t alignment)
 {
     return MapAlignedPagesLastDitch(size, alignment);
--- a/js/src/gc/Memory.h	2015-09-17 13:12:18.000000000 -0700
+++ b/js/src/gc/Memory.h	2017-07-06 15:07:27.136512238 -0700
@@ -43,7 +43,7 @@
 // Deallocate memory mapped content.
 void DeallocateMappedContent(void* p, size_t length);
 
-void* TestMapAlignedPagesLastDitch(size_t size, size_t alignment);
+static void* TestMapAlignedPagesLastDitch(size_t size, size_t alignment);
 
 } // namespace gc
 } // namespace js
